api/src/main/java/org/infinispan/api/async/AsyncCache.java

```diff
@@ -169,18 +459,34 @@
 
    /**
     * Register a cache listener with the supplied {@link CacheListenerOptions}
+    *
     * @param options
-    * @param types one or more {@link CacheEntryEventType}
+    * @param types   one or more {@link CacheEntryEventType}
     * @return
     */
    Flow.Publisher<CacheEntryEvent<K, V>> listen(CacheListenerOptions options, CacheEntryEventType... types);
 
    /**
-    * Process all entries using the supplied task
+    * Process entries using the supplied task
+    *
+    * @param task
+    * @return
+    */
+   default CompletionStage<Void> process(Set<K> keys, EntryProcessorTask<K, V> task) {
+      return process(keys, task, CacheOperationOptions.DEFAULT);
+   }
+
+   /**
+    * Process entries using the supplied task
+    *
     * @param task
+    * @param options
     * @return
     */
-   CompletionStage<Void> process(EntryConsumerTask<K, V> task);
+   CompletionStage<Void> process(Set<K> keys, EntryProcessorTask<K, V> task, CacheOperationOptions options);
```

Comment:
aren't we missing a `processAll` or something like java streams to do something in all keys?

Reply:
This is possible via `process(String query, EntryProcessorTask<K,V>)` by passing in the wildcard `%` for the query. But I agree an explicit API is better.

Reply:
I have added `processAll`